### DISCUSSION TITLE
Leopard Template: Blank Project: adding a new sprite

### DISCUSSION THREAD

**Discussion by sfederici** [leopard-js/leopard-mentors#7] [Cleanup Crew]
### Scratch Project URL

https://scratch.mit.edu/projects/1019731531

### Leopard Project URL

https://leopardjs.com/projects/oZ9X8OmH0J

### What kind of error(s) are you experiencing?

- [ ] Leopard is failing to generate JavaScript code
- [ ] The JavaScript code generated by Leopard doesn't work correctly
- [ ] The JavaScript code generated by Leopard is confusing or needs improvement
- [X] Other

### Description of issue(s)

I tried to add a new sprite. At first I tried adding an image (as I would do in Scratch) but I got a new tab in the currently selected sprite that shows the image when selected. First question: how would that image be related to the sprite?

Then I noticed that Leopard sprites are folders, so I tried to add a new folder. I gave it a name, but when I press enter or click outside the name box the folder disappears.

REVISION: the folder does not disappear. It just moves before the Stage folder. That is misleading to me. If we don't want to mix sprites and stage, maybe the stage could be placed as the first element of the Project Files area, so that when the new sprite folder name is confirmed the sprite doesn't move around?

**Comment by towerofnix** [comment]
Thank you for your feedback!

I think that the biggest confusion here is that Leopard projects actually have a pretty specific file-folder structure to them, and the Leopard editor does not make that structure very obvious — or help you to easily follow the structure, yourself.

As you've seen, sprites are folders in Leopard. Creating a new folder to start a new sprite (let's call it Sprite2) is the right thing to do. But, the folder really is just a folder - one that Leopard shows you in the sprite list, but that's it. There isn't anything "special" about the folder per se.

Of course, folders can contain folders. If you go to Sprite1 you will see that the "costumes" and "sounds" tabs actually have folder icons. When you clicked "Upload files" to add your costume, well, it *did* add the costume — but it added it right into the Sprite2 folder! This made it show up as a tab.

The right thing to do is to make a "costumes" folder *inside* of Sprite2. This will make a "costumes" tab show up. Then click this folder's "Create" -> "Upload files" button. Choose the image you want and Leopard will add it into the "costumes" folder.

Then you associate your actual sprite by doing the normal code, probably copy-pasting from Sprite1 or another sprite as the example. Replace the existing costumes with similar code, including the name of the file you uploaded in the "costumes" folder, and it should work. (The `x` and `y` values are the origin, or rotation center, of the costume, which you can customize to be half the image's dimensions if you want, or some other value.)

---

As a point of my own feedback, I think that maybe Leopard should have a "Sprite" option under Project Files' "Create" button. I've brought it up for discussion in this issue: PullJosh/leopard-website#52

**Comment by towerofnix** [comment]
> REVISION: the folder does not disappear. It just moves before the Stage folder. That is misleading to me. If we don't want to mix sprites and stage, maybe the stage could be placed as the first element of the Project Files area, so that when the new sprite folder name is confirmed the sprite doesn't move around?

Thanks, I've filed this to find or implement a solution, too: https://github.com/PullJosh/leopard-website/issues/53
